lib/seafoam/passes/truffle_translators/translators.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require "set"
+
+module Seafoam
+  module Passes
+    module TruffleTranslators
+      autoload :Default, "seafoam/passes/truffle_translators/default"
+      autoload :TruffleRuby, "seafoam/passes/truffle_translators/truffleruby"
+
+      TRUFFLE_LANGUAGES = {
+        "org.truffleruby" => "TruffleRuby",
+      }
+
+      class << self
+        def get_translator(node)
+          translator = node.visit_outputs(:bfs) do |entry|
+            declaring_class = (entry.props["code"] || entry.props.dig(
+              "nodeSourcePosition",
+              :method,
+            ))&.[](:declaring_class)
```

Comment:
This looks quite cryptic and is hard to parse.
How about:
```suggestion
            declaring_class = entry.props["code"] || entry.props.dig("nodeSourcePosition", :method)
            declaring_class = declaring_class[:declaring_class] if declaring_class
```